.github/workflows/get_build_targets.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# This limits the directory that bazel query is going to search under.
+readonly SEARCH_FOLDER="//source/common/..."
+
+set -e -o pipefail
+
+function get_targets() {
+  # Comparing the PR HEAD with the upstream master HEAD.
+  git diff --name-only HEAD FETCH_HEAD | while IFS= read -r line
+  do
+    # Only targets under those folders.
+    case "$line" in
+      source/*|include/*)
+        bazel query "rdeps($SEARCH_FOLDER, $line, 1)" 2>/dev/null
+        ;;
+    esac
+    # This chain of commands from left to right are:
+    # 1. Excluding the redundant .cc/.h targets that bazel query emits.
+    # 2. Storing only the unique output.
+    # 3. Limiting to the first 10 targets.
+  done | grep -v '.cc\|.h' | sort -u | head -n 10
```

Comment:
LGTM, but I think we need to make this `'\.cc\|\.h` or we risk matching too broadly.

Reply:
Not sure if I can commit that now?